experiment/simulation/js/display.js

```diff
@@ -0,0 +1,148 @@
+import { printErrors } from "./gate.js";
+import { registerGate } from "./main.js";
+'use strict';
+export let display = null;
+
+export function clearDisplay() {
+    display = null;
+}
+
+class Display {
+    constructor() {
+        this.id = "Display-" + window.numComponents++;
+        this.inputs = {
+            a: null,
+            b: null,
+            c: null,
+            d: null,
+            e: null,
+            f: null,
+            g: null
+        }
+        this.inputPoints = [];
+    }
+
+    generateComponent() {
+        const component = `<div id=${this.id} class="display"> <div class="display-container display-size-12 display-no"> <div id=${this.id}-segment-a class="segment-x segment-a segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-b class="segment-y segment-b segment-off"><span class="segment-border"></span></div> <div id=${this.id}-segment-c +' class="segment-y segment-c segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-d class="segment-x segment-d segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-e class="segment-y segment-e segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-f class="segment-y segment-f segment-off"><span class="segment-border"></span></div> <div id= ${this.id}-segment-g class="segment-x segment-g segment-off"><span class="segment-border"></span></div> </div></div>`;

```

Comment:
Is it possible to make it a bit more readable? Maybe put it in a separate file as a template and then import and do the binding?
Just see if there's a quick solution available. Do not waste too much time.

Reply:
Just moved it to new lines, and added indentation, looks like a HTML block now.